mmengine/runner/log_processor.py

```diff
@@ -236,12 +254,42 @@ def get_log_after_epoch(self,
         cur_loop = self._get_cur_loop(runner, mode)
         dataloader_len = len(cur_loop.dataloader)
 
-        custom_cfg_copy = self._parse_windows_size(runner, batch_idx)
+        custom_cfg_copy = copy.deepcopy(self.custom_cfg)
+        # remove prefix
+        custom_keys = [
+            self._remove_prefix(cfg['data_src'], f'{mode}/')
+            for cfg in custom_cfg_copy
+        ]
+
+        # Count the averaged time and data_time by epoch
+        if 'time' not in custom_keys:
+            custom_cfg_copy.append(
+                dict(
+                    data_src=f'{mode}/time',
+                    window_size='epoch',
+                    method_name='mean'))
+        if 'data_time' not in custom_keys:
+            custom_cfg_copy.append(
+                dict(
+                    data_src=f'{mode}/data_time',
+                    window_size='epoch',
+                    method_name='mean'))
+        parsed_cfg = self._parse_windows_size(runner, batch_idx,
+                                              custom_cfg_copy)
         # tag is used to write log information to different backends.
-        tag = self._collect_scalars(custom_cfg_copy, runner, mode)
+        ori_tag = self._collect_scalars(parsed_cfg, runner, mode,
+                                        not self.log_with_hierarchy)
         non_scalar_tag = self._collect_non_scalars(runner, mode)
-        tag.pop('time', None)
-        tag.pop('data_time', None)
+        # move `time` or `data_time` to the end of the log
+        tag = OrderedDict()
+        time_tag = OrderedDict()
+        # pop `time` and `data_time` to log them at last.
+        for key, value in ori_tag.items():
+            if 'time' not in key:
```

Comment:
May have potential bugs. E.g. I write a log called "altimeter", it will be put into the time_tag.